TodoList/sources/TodoList.Services/Nodes/CreateNodeService.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Threading.Tasks;
+using TodoList.Contracts.Models;
+using TodoList.Contracts.Repository;
+using TodoList.Contracts.Services;
+
+namespace TodoList.Services.Nodes
+{
+    internal class CreateNodeService : ICreateNodeService
+    {
+        private readonly INodesRepository _repository;
+        private readonly IGenerateIdService _generateIdService;
+        private readonly ICurrentTimeService _timeService;
+
+        public CreateNodeService(INodesRepository repository, IGenerateIdService generateIdService, ICurrentTimeService timeService)
+        {
+            _repository = repository;
+            _generateIdService = generateIdService;
+            _timeService = timeService;
+        }
+
+        public async Task<NodeModel> CreateNodeAsync(NodeModel node)
+            => await CreateNodeAsync(node, _generateIdService.GenerateId());
+
+        private async Task<NodeModel> CreateNodeAsync(NodeModel node, Guid id)
```

Comment:
I can't see the point of this function. Is there a reason for the 2 CreateNodeAsync functions?